src/test/java/io/airlift/units/TestDataSize.java

```diff
@@ -32,28 +32,53 @@
 import static io.airlift.units.DataSize.succinctBytes;
 import static io.airlift.units.DataSize.succinctDataSize;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.offset;
-import static org.assertj.core.data.Percentage.withPercentage;
+import static org.assertj.core.api.Assertions.within;
 import static org.testng.Assert.assertEquals;
 
 public class TestDataSize
 {
     @Test
     public void testSuccinctFactories()
     {
-        assertEquals(succinctBytes(123), new DataSize(123, BYTE));
-        assertEquals(succinctBytes((long) (5.5 * 1024)), new DataSize(5.5, KILOBYTE));
-        assertEquals(succinctBytes(5 * 1024 * 1024), new DataSize(5, MEGABYTE));
+        DataSize oneHundred = DataSize.ofBytes(123);
+        assertEquals(succinctBytes(oneHundred.toBytes()), oneHundred);
+        assertEquals(succinctBytes(oneHundred.toBytes()).getUnit(), BYTE);
+
+        DataSize fiveMiBi = DataSize.ofBytes(5 * 1024 * 1024);
+        assertEquals(succinctBytes(fiveMiBi.toBytes()), fiveMiBi);
+        assertEquals(succinctBytes(fiveMiBi.toBytes()).getUnit(), MEGABYTE);
+    }
+
+    @Test
+    public void testDeprecatedSuccinctFactories()
+    {
+        assertEquals(succinctDataSize(123, BYTE), DataSize.of(123, BYTE));
+        assertEquals(succinctDataSize(123, BYTE).getUnit(), BYTE);
 
-        assertEquals(succinctDataSize(123, BYTE), new DataSize(123, BYTE));
         assertEquals(succinctDataSize((long) (5.5 * 1024), BYTE), new DataSize(5.5, KILOBYTE));
-        assertEquals(succinctDataSize(5 * 1024, KILOBYTE), new DataSize(5, MEGABYTE));
+        assertEquals(succinctDataSize((long) (5.5 * 1024), BYTE).getUnit(), KILOBYTE);
+
+        assertEquals(succinctDataSize(5 * 1024, KILOBYTE), DataSize.of(5, MEGABYTE));
+        assertEquals(succinctDataSize(5 * 1024, KILOBYTE).getUnit(), MEGABYTE);
+    }
+
+    @Test
+    public void testToBytesValueString()
+    {
+        DataSize oneByte = DataSize.ofBytes(1);
+        assertEquals(oneByte.toString(), oneByte.toBytesValueString(), "exact values match toString()");
+
+        for (DataSize.Unit unit : DataSize.Unit.values()) {
```

Comment:
Add a few tests with fixed expected output, so it's easy to see and verify the expected behavior:
```java
assertEquals(DataSize.of(8, MEGABYTE).toBytesValueString(), "8388608B");
```

Reply:
Added